driver/src/orderbook/streamed/updating_orderbook.rs

```diff
@@ -92,51 +85,67 @@ impl StableXOrderBookReading for UpdatingOrderbook {
 /// Returns Ok when exit_indicator is dropped.
 /// Returns Err if the stream ends.
 async fn update_with_events_forever(
+    web3: Web3,
+    contract: Arc<dyn StableXContract>,
     orderbook: Arc<Mutex<Orderbook>>,
     orderbook_ready: Arc<AtomicBool>,
     mut block_timestamp_reader: impl BlockTimestampReading,
     exit_indicator: oneshot::Receiver<()>,
-    past_events: impl Future<Output = Result<Vec<Event<batch_exchange::Event>>, ExecutionError>>,
-    stream: impl Stream<Item = Result<Event<batch_exchange::Event>, ExecutionError>>,
 ) -> Result<()> {
-    // `select!` requires the futures to be fused...
-    let exit_indicator = exit_indicator.fuse();
-    let past_events = past_events.fuse();
-    let stream = stream.fuse();
-    // ...and pinned.
-    pin_mut!(exit_indicator);
-    pin_mut!(past_events);
-    pin_mut!(stream);
+    const POLL_INTERVALL: Duration = Duration::from_secs(15);
+    const BLOCK_RANGE: u64 = 25;
 
     log::info!("Starting event based orderbook updating.");
 
+    // `select!` requires the futures to be fused and pinned.
+    let exit_indicator = exit_indicator.fuse();
+    pin_mut!(exit_indicator);
+
+    let mut last_handled_block = 0u64;
     loop {
-        // We select over everything together instead of for example the past events first then the
-        // stream to ensure that the stream gets polled at least once which it needs in order to
-        // create the corresponding filter on the node.
+        let current_block = web3.eth().block_number().compat().await?;
+        let from_block = last_handled_block.saturating_sub(BLOCK_RANGE);
```

Comment:
👍 Make sense. I was originally thinking to start from MIN(current - block_range, last_handled_block), but it's right that we might have missed a reorg that happened around last_handled_block.